src/gridfs/bucket.ts

```diff
@@ -66,12 +60,12 @@ export class GridFSBucket {
    *
    * Returns a Stream to which the application will write the contents.
    */
-  openUploadStreamWithId(
+  async openUploadStreamWithId(
     id: FileId,
     filename: string,
     options?: GridFSUploadOptions,
   ) {
-    if (!this.#checkedIndexes) this.#checkIndexes();
+    if (!this.#checkedIndexes) await this.#checkIndexes();
```

Comment:
This only should be the issue. And the functions referencing to openUploadStreamWithId

Reply:
Yes. I encountered this bug.